lib/websocket.js

```diff
@@ -117,22 +112,83 @@ function init (env, ctx, server) {
             return { result: 'Not permitted' };
           }
         }
-
-        if (!data._id) {
+        
+        if (action === 'dbUpdate' && !data._id) {
           console.log('WS dbUpdate call: ', 'Missing _id', data);
           return { result: 'Missing _id' };
         }
 
+        return null;
+      }
+
+      // dbUpdate message  
+      //  {
+      //    collection: treatments
+      //    _id: 'some mongo record id'
+      //    data: {
+      //      field_1: new_value,
+      //      field_2: another_value
+      //    }
+      //  }
+      socket.on('dbUpdate', function dbUpdate(data, callback) {
+       console.log(LOG_IO + 'dbUpdate client ID: ', socket.client.id, ' data: ', data);
+       var collection = supportedCollections[data.collection];
+        
+        var check = checkConditions('dbUpdate', data);
+        if (check) {
+         if (callback) {
+            callback( check );
+          }
+          return;
+        }
+        
         ctx.store.db.collection(collection).update(
           { '_id': new ObjectID(data._id) },
           { $set: data.data }
         );
           
-        //console.log(ret);
-        callback( { result: 'success' } );
+        if (callback) {
+          callback( { result: 'success' } );
+        }
         ctx.bus.emit('data-received');
       });
       
+      // dbAdd message  
+      //  {
+      //    collection: treatments
+      //    data: {
+      //      field_1: new_value,
+      //      field_2: another_value
+      //    }
+      //  }
+      socket.on('dbAdd', function dbAdd(data, callback) {
+       console.log(LOG_IO + 'dbAdd client ID: ', socket.client.id, ' data: ', data);
+        var collection = supportedCollections[data.collection];
+        
+        var check = checkConditions('dbAdd', data);
+        if (check) {
+         if (callback) {
+            callback( check );
+          }
+          return;
+        }
+        
+        if (data.collection === 'treatments' && !('eventType' in data.data)) {
+          data.data.eventType = '<none>';
+        }
+        if (!('created_at' in data.data)) {
+          data.data.created_at = new Date().toISOString();
+        }
+        
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: ["result" is defined but never used](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1725154834/issues/source?bid=2452379&fileBranchId=3052457#l183)